File: ilo.lua
local function read_file(filename)
    local file = io.open(filename, "r")
    if not file then
        return nil
    end
    local content = file:read("*a")
    file:close()
    return content
end

local function get_word_definition(data, word)
    local entries = data["data"]
    if entries then
        local entry = entries[word]
        if entry and entry["def"] and entry["def"]["en"] then
            return entry["def"]["en"]
        end
    end
    return nil
end

local function parse_json(data)
    local success, result = pcall(function() return load("return " .. data)() end)
    return success and result or nil
end

local function main()
    local filename = "data.txt"
    local data = read_file(filename)
    if not data then
        print("Error: Unable to read data from the file.")
        return
    end

    data = parse_json(data)
    if not data then
        print("Error: Invalid JSON data.")
        return
    end

    local word = "kepeken" -- Replace with the word you want to look up
    local definition = get_word_definition(data, word)
    if definition then
        print(string.format("The English definition of '%s' is: %s", word, definition))
    else
        print(string.format("No English definition found for '%s'.", word))
    end
end

main()

File: ilo.py
import requests
print("nimi seme?")
word = input()

def get_word_definition(word):
    url = "https://linku.la/jasima/data.json"
    response = requests.get(url)
    data = response.json()
    
    if "data" in data:
        for entry in data["data"].values():
            if entry["word"] == word:
                if "def" in entry:
                    if "en" in entry["def"]:
                        return entry["def"]["en"]
    
    return None

# Example usage
definition = get_word_definition(word)
if definition:
    print()
    print(word,":")
    print(definition,"(en)")
else:
    print("mi sona ala e '{word}' a!")
